crates/typst-library/src/model/table.rs

```diff
@@ -259,10 +263,37 @@ impl TableElem {
 }
 
 impl Show for Packed<TableElem> {
-    fn show(&self, engine: &mut Engine, _: StyleChain) -> SourceResult<Content> {
-        Ok(BlockElem::multi_layouter(self.clone(), engine.routines.layout_table)
-            .pack()
-            .spanned(self.span()))
+    fn show(&self, engine: &mut Engine, styles: StyleChain) -> SourceResult<Content> {
+        Ok(if TargetElem::target_in(styles).is_html() {
+            // TODO: This is a hack, it is not clear whether the locator is actually used by HTML.
+            // How can we find out whether locator is actually used?
+            let locator = Locator::root();
+            let elem = |tag, body| HtmlElem::new(tag).with_body(Some(body)).pack();
+            let grid = table_to_cellgrid(self, engine, locator, styles)?;
+            let rows = grid.entries.chunks(grid.cols.len()).map(|row| {
+                let row = row.iter().flat_map(|entry| entry.as_cell());
+                elem(tag::tr, Content::sequence(row.map(|cell| cell.body.clone())))
+            });
+            let mut rows: Vec<_> = rows.collect();
+
+            let footer = grid.footer.map(|ft| {
+                elem(tag::tfoot, Content::sequence(rows.drain(ft.unwrap().start..)))
+            });
+            let header = grid.header.map(|hd| {
+                elem(tag::thead, Content::sequence(rows.drain(..hd.unwrap().end)))
```

Comment:
Cells within headers should be `th` instead of `td`.

Reply:
What about cells within footers?

Reply:
Those are normal `td`s

Reply:
OK, this is now done in [b2b2593](https://github.com/typst/typst/pull/5666/commits/b2b2593832cc17282f3e41fa54dcb2923cbf7354). That change was really quite hairy ...